cmake/check_cuda.cmake

```diff
@@ -60,6 +60,7 @@ elseif(USE_CUDA)
 else()
   file(GLOB generator_cuda_srcs "${GENERATORS_ROOT}/*_cuda*.*")
   list(REMOVE_ITEM generator_srcs ${generator_cuda_srcs})
+  add_compile_definitions(USE_CUDA=0)
```

Comment:
Any code usage of ```#ifndef USE_CUDA``` would no longer work with this change, right?

Reply:
Correct, we don't have any usage like that as it'd be inconsistent. And this will prevent anyone from trying!